### ISSUE TITLE
Newest sharp version 0.31 can not attach icc color proifle to image format PNG anymore

### ISSUE THREAD

**Issue by xuanngoc** [lovell/sharp#3390]
### Is this a possible bug in a feature of sharp, unrelated to installation?

- [x] Running `npm install sharp` completes without error.
- [x] Running `node -e "require('sharp')"` completes without error.

If you cannot confirm both of these, please open an [installation issue](https://github.com/lovell/sharp/issues/new?labels=installation&template=installation.md) instead.

### Are you using the latest version of sharp?

- [x] I am using the latest version of `sharp` as reported by `npm view sharp dist-tags.latest`.

If you cannot confirm this, please upgrade to the latest version and try again before opening an issue.

If you are using another package which depends on a version of `sharp` that is not the latest, please open an issue against that package instead.

### What is the output of running `npx envinfo --binaries --system --npmPackages=sharp --npmGlobalPackages=sharp`?

```  
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 3.32 GB / 23.27 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.2.0 - ~/.nvm/versions/node/v18.2.0/bin/node
    npm: 8.9.0 - ~/.nvm/versions/node/v18.2.0/bin/npm
  npmPackages:
    sharp: ^0.31.0 => 0.31.0 
```
### What are the steps to reproduce?

create sharp -> extract icc buffer -> save file icc ->  create sharp -> resize -> withMetadata({icc: icc_path_file}) -> toBuffer

### What is the expected behaviour?

The output image format PNG embedded icc - color profile
### Please provide a minimal, standalone code sample, without other dependencies, that demonstrates this problem

```
const path = require("path");
const sharp = require('sharp');
const fs = require('fs');

const file = fs.readFileSync(path.resolve(__dirname, '../../images/metadata/fruit-g236fa3733_1920.png'));

sharp(file).metadata((err, data) => {
  console.log(data)
  const {icc} = data;

  fs.writeFileSync(path.resolve(__dirname, '../../images/metadata/fruit-g236fa3733_1920.icc'), icc)

  sharp(file)
    .resize(300, 300, {fit: 'inside'})
    .withMetadata({icc: path.resolve(__dirname, '../../images/metadata/fruit-g236fa3733_1920.icc')})
    .toBuffer((err, data) => {
      console.log(err);
      fs.writeFileSync(path.resolve(__dirname, '../../images/metadata/fruit-g236fa3733_1920_300_icc.png'), data);
    }
  )
})
```
### Please provide sample image(s) that help explain this problem

Input image
![fruit-g236fa3733_1920](https://user-images.githubusercontent.com/32450324/192720507-f5c970fa-c703-4e02-9774-ed283e7f66af.png)

Output image
![fruit-g236fa3733_1920_300_icc](https://user-images.githubusercontent.com/32450324/192720714-6bf2ed21-9182-41f7-82d0-591d4e303aee.png)

**Comment by xuanngoc** [comment]
Hello, I got issue with with newest version 0.31. As I can see in changelog, there are no different about image png, metadata or icc. I guess this issue from libvips version that change a lot about libpng, ...I remember that in previous sharp version, image png still can attach icc, but with ver 0.31 it can not. Please let me know if possible. Thank you.

**Comment by xuanngoc** [comment]
I tested with `exiftool` to extract icc from the input image then embedded it to the output image. It worked!. I guess that somethings wrong when sharp update new Libvips version.

**Comment by xuanngoc** [comment]
Btw a good thing in new version is the PNG images have better quality than before.

**Comment by letuanvu59** [comment]
I have the same problem with the new version 0.31

**Comment by letuanvu59** [comment]
@lovell lo
Thanks for the quick support, I see libvips already make a new version so when sharp makes a new version?

**Comment by lovell** [comment]
sharp v0.31.1 now available with libvips v8.13.2
